### PR TITLE
[WS2_32_WINETEST] Patch ws2_32:sock wine test to run on ReactOS

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Purpose

_Disable ReactOS specific tests which are not implemented yet._

JIRA issue: [ROSTESTS-385](https://jira.reactos.org/browse/ROSTESTS-385)

## Proposed changes

_Bypass part of test that uses MEM_WRITE_WATCH._
_Bypass three tests that are unimplemented: 1) WSPAcceptEx, 2) WSPConnectEx, and 3) WSPDisconnectEx_

Testbot results:
winetest_ws2_32_fix_05.patch JID66261 on top of 0.4.15-dev-6526-g8d35887

VBox: https://reactos.org/testman/compare.php?ids=89111,89117 LGTM
KVM:  https://reactos.org/testman/compare.php?ids=89110,89118 LGTM

### CODE REVIEW

File: modules/rostests/winetests/ws2_32/sock.c [hunk 1]
```diff
@@ -7010,6 +7010,15 @@ static void test_write_watch(void)
     base = VirtualAlloc( 0, size, MEM_RESERVE | MEM_COMMIT | MEM_WRITE_WATCH, PAGE_READWRITE );
     ok( base != NULL, "VirtualAlloc failed %u\n", GetLastError() );
 
+#ifdef __REACTOS__  /* ROSTESTS-385 */
+    if (!base)
+    {
+        /* MEM_WRITE_WATCH is not supported yet in ReactOS */
+        win_skip( "VirtualAlloc not supported\n" );
```
Comment (SergeGautherie) [inline modules/rostests/winetests/ws2_32/sock.c:7017]:
```suggestion
        win_skip("VirtualAlloc(MEM_WRITE_WATCH) is not supported yet on ReactOS\n");
```

File: modules/rostests/winetests/ws2_32/sock.c [hunk 2]
```diff
@@ -11620,9 +11631,13 @@ START_TEST( sock )
     test_GetAddrInfoW();
     test_GetAddrInfoExW();
     test_getaddrinfo();
+
+#ifndef __REACTOS__    /* ROSTESTS-385 */
+    /* UNIMPLEMENTED in ReactOS See dll/win32/msafd/misc/stubs.c */
```
Comment (SergeGautherie) [inline modules/rostests/winetests/ws2_32/sock.c:11636]:
```suggestion
#ifdef __REACTOS__
    win_skip("WSPAcceptEx(), WSPConnectEx() and WSPDisconnectEx() are UNIMPLEMENTED on ReactOS\n");
#else
```

File: modules/rostests/winetests/ws2_32/sock.c [hunk 3]
```diff
@@ -8826,6 +8836,7 @@ static void test_DisconnectEx(void)
     closesocket(connector);
     closesocket(listener);
 }
+#endif // ifndef __REACTOS__   /* ROSTESTS-385 */
```
Comment (SergeGautherie) [inline modules/rostests/winetests/ws2_32/sock.c:8839]:
```suggestion
#endif // __REACTOS__
```

File: modules/rostests/winetests/ws2_32/sock.c [hunk 4]
```diff
@@ -8012,6 +8021,7 @@ static void test_getaddrinfo(void)
     }
 }
 
+#ifndef __REACTOS__    /* ROSTESTS-385 */
```
Comment (SergeGautherie) [inline modules/rostests/winetests/ws2_32/sock.c:8024]:
```suggestion
#ifndef __REACTOS__
```

File: modules/rostests/winetests/ws2_32/sock.c [hunk 5]
```diff
@@ -7010,6 +7010,15 @@ static void test_write_watch(void)
     base = VirtualAlloc( 0, size, MEM_RESERVE | MEM_COMMIT | MEM_WRITE_WATCH, PAGE_READWRITE );
     ok( base != NULL, "VirtualAlloc failed %u\n", GetLastError() );
 
+#ifdef __REACTOS__  /* ROSTESTS-385 */
```
Comment (SergeGautherie) [inline modules/rostests/winetests/ws2_32/sock.c:7013]:
```suggestion
#ifdef __REACTOS__
```

File: modules/rostests/winetests/ws2_32/sock.c [hunk 6]
```diff
@@ -11620,9 +11631,13 @@ START_TEST( sock )
     test_GetAddrInfoW();
     test_GetAddrInfoExW();
     test_getaddrinfo();
+
+#ifndef __REACTOS__    /* ROSTESTS-385 */
+    /* UNIMPLEMENTED in ReactOS See dll/win32/msafd/misc/stubs.c */
     test_AcceptEx();
```
Comment (ThFabba) [inline modules/rostests/winetests/ws2_32/sock.c:11637]: We usually skip these using `if (!winetest_interactive)`, that way you also don't need to ifdef out the functions

Comment (Doug-Lyons) [reply modules/rostests/winetests/ws2_32/sock.c:11647]: Changed. Thanks.

File: modules/rostests/winetests/ws2_32/sock.c (hunk 1 above)
Comment (Doug-Lyons) [reply modules/rostests/winetests/ws2_32/sock.c]: Changed. Thanks.

File: modules/rostests/winetests/ws2_32/sock.c (hunk 2 above)
Comment (Doug-Lyons) [reply modules/rostests/winetests/ws2_32/sock.c]: Changed. Thanks.

**Review by JoachimHenze — APPROVED** [review]

File: modules/rostests/winetests/ws2_32/sock.c [hunk 7]
```diff
@@ -11620,9 +11629,25 @@ START_TEST( sock )
...
+    {
+        skip("WSPAcceptEx(), WSPConnectEx() and WSPDisconnectEx() are UNIMPLEMENTED on ReactOS\n");
+        skip("Skipping tests due to hang. See ROSTESTS-385\n");
+    }
+    else
+    {
+        /* UNIMPLEMENTED in ReactOS See dll/win32/msafd/misc/stubs.c */
+        test_AcceptEx();
+        test_ConnectEx();
+        test_DisconnectEx();
```
[the remark this answers is not in the document]
Comment (Doug-Lyons) [reply modules/rostests/winetests/ws2_32/sock.c:11644]: I followed the pattern in winhttp:winhttp test as shown here:
```
#ifdef __REACTOS__
    if (!winetest_interactive)
    {
        skip("Skipping tests due to hang. See ROSTESTS-350\n");
    }
    else
    {
        test_multiple_reads(si.port);
        test_cookies(si.port);
        test_request_path_escapes(si.port);
        test_passport_auth(si.port);

        /* send the basic request again to shutdown the server thread */
        test_basic_request(si.port, NULL, quitW);
    }
#else
    test_multiple_reads(si.port);
    test_cookies(si.port);
    test_request_path_escapes(si.port);
    test_passport_auth(si.port);

    /* send the basic request again to shutdown the server thread */
    test_basic_request(si.port, NULL, quitW);
#endif
```

File: modules/rostests/winetests/ws2_32/sock.c [hunk 8]
```diff
@@ -11620,9 +11629,25 @@ START_TEST( sock )
     test_GetAddrInfoW();
     test_GetAddrInfoExW();
     test_getaddrinfo();
+
+#ifdef __REACTOS__
+    if (!winetest_interactive)
+    {
+        skip("WSPAcceptEx(), WSPConnectEx() and WSPDisconnectEx() are UNIMPLEMENTED on ReactOS\n");
+        skip("Skipping tests due to hang. See ROSTESTS-385\n");
```
Comment (mm1044) [inline modules/rostests/winetests/ws2_32/sock.c:11637]: I think this second line belongs to commit message, or at least a comment. At both places.
```suggestion
```

File: modules/rostests/winetests/ws2_32/sock.c [hunk 9]
```diff
@@ -11620,9 +11629,25 @@ START_TEST( sock )
...
+    }
+    else
+    {
+        /* UNIMPLEMENTED in ReactOS See dll/win32/msafd/misc/stubs.c */
```
Comment (mm1044) [inline modules/rostests/winetests/ws2_32/sock.c:11641]:
```suggestion
```

File: modules/rostests/winetests/ws2_32/sock.c (hunk 8 above)
Comment (SergeGautherie) [reply modules/rostests/winetests/ws2_32/sock.c:11637]: I think this second line belongs to commit message, or at least a comment. At both places.
```suggestion
```

File: modules/rostests/winetests/ws2_32/sock.c (hunk 9 above)
Comment (SergeGautherie) [reply modules/rostests/winetests/ws2_32/sock.c:11641]:
```suggestion
```

**Comment by JoachimHenze** [comment]
My recommendation would be to **first** disable or fix those parts of ws2:sock that do fail even on WHS bot currently, and then see what else we do need on top of that to make the suite run ok on ros:
https://reactos.org/testman/compare.php?ids=89609

**Review by tkreuzer — APPROVED** [review]

File: modules/rostests/winetests/ws2_32/sock.c (hunk 7 above)
[the remark this answers is not in the document]
Comment (tkreuzer) [reply modules/rostests/winetests/ws2_32/sock.c:11644]: You could do
```
#ifdef __REACTOS__
    if (!winetest_interactive)
    {
        skip("Skipping tests due to hang. See ROSTESTS-350\n");
    }
    else
    {
#endif
    test_multiple_reads(si.port); test_cookies(si.port);
    test_request_path_escapes(si.port);
    test_passport_auth(si.port);
    /* send the basic request again to shutdown the server thread */
    test_basic_request(si.port, NULL, quitW);
#ifdef __REACTOS__
    }
#endif

[the remark this answers is not in the document]
Comment (Doug-Lyons) [reply modules/rostests/winetests/ws2_32/sock.c]: Changed as suggested. Thanks.

**Review by HBelusca — APPROVED** [review]

**Comment by Doug-Lyons** [comment]
> My recommendation would be to **first** disable or fix those parts of ws2:sock that do fail even on WHS bot currently, and then see what else we do need on top of that to make the suite run ok on ros: https://reactos.org/testman/compare.php?ids=89609

The number of tests run varies quite a lot as you can see here:
https://reactos.org/testman/compare.php?ids=92578,92730,92744,92749,92779

After we have a few successful runs on ReactOS, I will see what we can do about these. I suspect that the same 3 tests are the ones that fail consistently. If so, then I will see what can be done to either fix them or skip them. Thanks.

**Comment by Doug-Lyons** [comment]
Thanks, @HBelusca for your approval and the merge. Thanks, @tkreuzer and @JoachimHenze for your approvals.

**Pull request merged**